src/nonlinear/nonlinear.jl

```diff
@@ -0,0 +1 @@
+#TODO: Add support for general nonlinear problems
```

Comment:
I would remove this file and wait for the implementation of the nonlinear extension